GVFS/GVFS.Common/Paths.Shared.cs

```diff
@@ -68,5 +69,28 @@ public static string ConvertPathToGitFormat(string path)
         {
             return path.Replace(Path.DirectorySeparatorChar, GVFSConstants.GitPathSeparator);
         }
+
+        public static IEnumerable<string> GetFilesRecursive(string rootDirectory)

```

Comment:
I would rename this to `GetRecursiveRelativeFilePaths` - the fact that they're relative is quite important here, and it's returning paths, not files.